src/app/loadout-builder/generated-sets/CompareLoadoutsDrawer.tsx

```diff
@@ -232,6 +232,7 @@ export default function CompareLoadoutsDrawer({
               <Select<Loadout>
                 key="select-loadout"
                 value={selectedLoadout}
+                maxDropdownHeight={200}
```

Comment:
I don't think we should hardcode this - it should be dynamic based on the distance between the select element and the bottom of the viewport. That's why I hadn't done this yet :-)

Linking #8374

Reply:
Yeah I wasn't super happy with this but it was quick to implement and get some realy feedback. I can certainly make this dynamic based on select element + viewport bottom.

Reply:
ok, so I took another pass. there's a bit of code that's already doing something similar for the Search Results dropdown that I think we could standardise into a hook:

```
  // Determine a maximum height for the results menu
  useEffect(() => {
    if (menuRef.current && window.visualViewport) {
      const { y, height } = menuRef.current.getBoundingClientRect();
      const { height: viewportHeight } = window.visualViewport;
      // pixels remaining in viewport minus offset minus 10px for padding
      const pxAvailable = viewportHeight - y - height - 10;
      const itemHeight = // TBC ;

      // constrain to size that would allow only whole items to be seen
      setDropdownHeight(Math.floor(pxAvailable / itemHeight) * itemHeight);
    }
  }, [...]);
```

Using this I can make it dynamic and it will neatly sit below the button on the row, with somebreathing room: 

![Screenshot 2023-05-12 at 17 37 46](https://github.com/DestinyItemManager/DIM/assets/10519003/96aefdd9-13b6-427c-a7fb-dad9645f1c56)

(Again, this just scroll I just can't screenshot it)

What do you think? Or would it be better to tackle the migration to Floating UI?

Reply:
That looks great!

Reply:
OK, I've created the new hook which you can see here: https://github.com/DestinyItemManager/DIM/pull/9449/commits/8655483d4879fbe84e59af85edf86e100fe4101c#diff-38c1690a9238e25e48c3296475be4658019a98b0a339b077c224a6ab8f55318eR106, I'll look at adding a test to this now given it's a common hook. 

There's still a magic number used as we don't know the item height until it renders (which we use to set a height that's an whole number of items - similar behaviour to the `SearchBar`). That said I think this works pretty well for this case and the other usecases of the Select doesn't appear to have height limitations (`ItemTagSelection`, `CustomStatsSetting` and `SettingPage`) which all have limited, non user-defined, options.

The other thing that could be good in a followup is move the `SearchBar` over to this impl, I noticed it also doesn't scroll if the dropdown is truncated.